### PR TITLE
Changing from workflow to action

### PR DESCRIPTION
**Pull request by PaurushGarg**
*Issue #, if available:*

*Description of changes:*
PR to fix this error:https://github.com/aws-observability/aws-otel-java-instrumentation/actions/runs/7426995888/job/20211834758

Seems like a workflow cannot be invoked from inside a step (https://github.com/actions/checkout/issues/692#issuecomment-1259789797) , so changing from workflow to action (to allow invocation from step instead of job).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by bryan-aguilar** [comment]
It's because you created a [`composite`](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action) action not a [`reusable`](https://docs.github.com/en/actions/using-workflows/reusing-workflows) workflow.

**Review by rapphil — APPROVED** [review]

**Review by bryan-aguilar — APPROVED** [review]

File: .github/actions/cpUtility-testing/action.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v3
+
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v3
+
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v4
+      with:
+        role-to-assume: ${{ secrets.JAVA_INSTRUMENTATION_SNAPSHOT_ECR }}
```
Comment (bryan-aguilar) [inline .github/actions/cpUtility-testing/action.yml:32]: Does this role have the correct access to both ecr repos?

File: .github/actions/cpUtility-testing/action.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      run: .github/scripts/test-adot-javaagent-image.sh "${{ inputs.image_uri_with_tag }}" "${{ inputs.adot-java-version }}"
+
+    - name: Build and push image
+      uses: docker/build-push-action@v5
+      with:
+        push: true
+        build-args: "ADOT_JAVA_VERSION=${{ inputs.adot-java-version }}"
+        context: .
+        platforms: linux/amd64,linux/arm64
+        tags: ${{ inputs.image_uri_with_tag }}
```
Comment (vasireddy99) [inline .github/actions/cpUtility-testing/action.yml:63]: nit - new line.

**Review by vasireddy99 — APPROVED** [review]

File: .github/actions/cpUtility-testing/action.yml (hunk 1 above)
Comment (PaurushGarg) [reply .github/actions/cpUtility-testing/action.yml:32]: yes, it has: https://github.com/aws-observability/aws-otel-ops/blob/4859e9eadade7088b4801992344409bbf2fa5b3a/iam-role-setup/lib/config/iam/integ-test-java-config.ts#L28

**Pull request merged**